Modules/IO/DCMTK/test/itkDCMTKGetDicomTagsTest.cxx

```diff
@@ -0,0 +1,157 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkImageFileReader.h"
+#include "itkImageFileWriter.h"
+#include "itkDCMTKFileReader.h"
+#include "itkTestingMacros.h"
+
+#include <fstream>
+
+int
+itkDCMTKElementGetTest(int ac, char * av[])
+{
+
+  if (ac != 2)
+  {
+    std::cerr << "Missing Parameters." << std::endl;
+    std::cerr << "Usage: " << itkNameOfTestExecutableMacro(av) << " DicomImage" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  constexpr unsigned int Dimension = 3;
+  using InputPixelType = unsigned short;
+  using InputImageType = itk::Image<InputPixelType, Dimension>;
+  using ReaderType = itk::ImageFileReader<InputImageType>;
+
+  itk::DCMTKFileReader fileReader;
+  fileReader.SetFileName(av[1]);
+  fileReader.LoadFile();
+
+  bool throwException = false;
+
+  std::string strTarget;
+
+  // Study Date
+  unsigned short group = 0x0008;
+  unsigned short element = 0x0021;
+  if (fileReader.GetElementDA(group, element, strTarget, throwException) == EXIT_SUCCESS)
+  {
+    // YYYYMMDD
+    // Expect 20030625
+    std::cout << strTarget << std::endl;
+  }
+
+  // Study Time
+  group = 0x0008;
+  element = 0x0030;
+  // seq.GetElementTM(group, element, tag)
+  if (fileReader.GetElementTM(group, element, strTarget, throwException) == EXIT_SUCCESS)
+  {
+    // Expect 152734
+    std::cout << strTarget << std::endl;
+  }
+
+  // Patient Name
+  // Expect "Wes Turner"
+  group = 0x0010;
+  element = 0x0010;
+  if (fileReader.GetElementPN(group, element, strTarget, throwException) == EXIT_SUCCESS)
+  {
+    std::cout << strTarget << std::endl;
+  }
+
+  // Patient Sex
+  // Expect "O"
+  group = 0x0010;
+  element = 0x0040;
+  if (fileReader.GetElementCS(group, element, strTarget, throwException) == EXIT_SUCCESS)
+  {
+    std::cout << strTarget << std::endl;
+  }
+
+  // Patient Weight
+  // Expect 68.039
+  group = 0x0010;
+  element = 0x1030;
+  if (fileReader.GetElementDS(group, element, 1, &strTarget, throwException) == EXIT_SUCCESS)
+  {
+    std::cout << strTarget << std::endl;
+  }
+
+  // Echo Number
+  group = 0x0018;
+  element = 0x0086;
+  // Expect (12576)"1"
+  itk::int32_t int32Target;
+  if (fileReader.GetElementIS(group, element, int32Target, throwException) == EXIT_SUCCESS)
+  {
+    std::cout << int32Target << std::endl;
+  }
+
+  // Acquisition Duration
+  group = 0x0019;
+  element = 0x105a;
+  // Expect 1304791694
```

Comment:
Strange, my viewer and also _dcmdump_ show
`(0019,105a) FL 414273984                                #   4, 1 AcquisitionDuration`
not 1304791694. Both are somehow strange for float, there may be an issue in the file.
Also applicable to private tags in the file, it is correct to access private tags (odd group) using private creator, not just by group and element. But to simply increase coverage by using the somehow it is probably not important.

Reply:
Not sure where I got the value from :-S. I probably used an online viewer, which might not be fully trustworthy.

I am not able to build the module locally, so I was relying on an online tool rather than installing locally some library/tool. But I am willing to fix it if anybody can point me to a dependable tool that I could use to potentially get the accurate values for more tags.

Reply:
> Also applicable to all private tags in the test, it is correct to access private tags (odd group) using private creator, not just by group and element. 

Can you elaborate on this? If there is a better way to access the tags, I may have a look to also add them (?).

Reply:
> Can you elaborate on this? If there is a better way to access the tags, I may have a look to also add them (?).

Sorry, not in the near future. The test file has explicit transfer syntax, it works, a VR is known. Imagine some implicit file, VR can be taken only from a private dictionary, otherwise only OB could be used as workaround.

Reply:
This particular tag is also known for DCMTK (is in private dictionary), so it is OK.

Reply:
> I am not able to build the module locally, so I was relying on an online tool rather than installing locally some library/tool.

I have no problem with Linux build and internal DCMTK. I remember with external and at least static libraries there were problems too, something with missing libraries (they were there, but not found).

P.S.
I have to admit I am not experienced with DCMTK, actually I am testing it for PACS stuff, it seems to be clean and well maintained library.